### PR TITLE
Add more tests for the GoogleCloudExtensionPackage.

### PR DESCRIPTION
**Pull request by ILMTitan**
These were pulled from the Google Cloud Explorer Options page PR #857.

### CODE REVIEW

File: GoogleCloudExtension/GoogleCloudExtension/Analytics/EventsReporterWrapper.cs [hunk 1]
```diff
@@ -64,24 +68,23 @@ public static void EnsureAnalyticsOptIn()
...
-        {
-            s_reporter.Value?.ReportEvent(
-                source: ExtensionEventSource,
-                eventType: ExtensionEventType,
-                eventName: eventData.Name,
-                userLoggedIn: CredentialsStore.Default.CurrentAccount != null,
-                projectNumber: CredentialsStore.Default.CurrentProjectNumericId,
-                metadata: eventData.Metadata);
-        }
+        public static Action<AnalyticsEvent> ReportEvent { get; internal set; } = (AnalyticsEvent eventData) =>
```
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension/Analytics/EventsReporterWrapper.cs]: This is hacky, not easy to follow. 
May you instead mock the IEventsReporter inside by changing the CreateReporter method? 
Or create an internal  SetReporter, ChangeReporter method?

Comment (ILMTitan) [reply GoogleCloudExtension/GoogleCloudExtension/Analytics/EventsReporterWrapper.cs]: Done.

**Comment by ILMTitan** [comment]
PTAL

File: GoogleCloudExtension/GoogleCloudExtension/Analytics/EventsReporterWrapper.cs [hunk 2]
```diff
@@ -64,17 +68,16 @@ public static void EnsureAnalyticsOptIn()
...
         {
-            s_reporter = new Lazy<IEventsReporter>(CreateReporter);
+            ReporterLazy = new Lazy<IEventsReporter>(CreateReporter);
         }
 
         /// <summary>
         /// Called to report an interesting event to analytics. If there's a queue of events it will be
         /// flushed as well.
         /// </summary>
-        /// <param name="eventData"></param>
```
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension/Analytics/EventsReporterWrapper.cs]: Add this back

**Pull request merged**